lms/lmsweb/views.py

```diff
@@ -104,7 +105,19 @@ def get_next_url(url_next_param: Optional[str]):
     return redirect(next_url or url_for('main'))
 
 
+@webapp.errorhandler(429)
+def ratelimit_handler(e):
+    log.info(f'IP Address: {get_remote_address()}: {e}')
+    return make_response(
+        jsonify(error='ratelimit exceeded %s' % e.description), 429,
+    )
+
+
 @webapp.route('/login', methods=['GET', 'POST'])
+@limiter.limit(
+    '5/minute;50/hour',
+    deduct_when=lambda response: response.status_code != 200,
```

Comment:
I think we always return 200, even when the login failed.

Reply:
Tried it and it works in this way

Reply:
You've checked that many refreshes to the login pages doesn't block the user?

Reply:
You're right.
And yes, I checked it